crypto/data_formats.go

```diff
@@ -53,3 +55,66 @@ func HashLogRoot(root trillian.SignedLogRoot) ([]byte, error) {
 	}
 	return hash[:], nil
 }
+
+// LogRootV1 contains the fields verified by SignedLogRoot
+type LogRootV1 struct {
+	Version        uint16
```

Comment:
IIUC this field will always have value 1 (i.e. `LogRootV1.Version==LOG_ROOT_FORMAT_V1` is an invariant)?  If so, a comment to that effect might be helpful.

Also, if this is effectively an enum, it might be worth using the `tls.Enum` type (with a `size:2` tag).  

More generally, I wonder whether using [variants](https://tools.ietf.org/html/rfc5246#section-4.6.1) might be more future proof -- e.g. something in TLS-speak like:
```
enum { v1(1), (65535)} Version;
struct {
  uint64 tree_size;
  opaque root_hash<0..128>;
  uint64 timestamp_nanos;
  uint64 revision;
  opaque metadata<0..65535>;
} LogRootV1;
struct {
  Version version;
  select(version) {
    case v1: LogRootV1;
  }
} LogRoot;
```
is then easily extendable to:
```
enum { v1(1), v2(2), (65535)} Version;
...
struct {
  uint64 tree_size;
  opaque root_hash<0..128>;
  uint64 timestamp_nanos;
  uint64 revision;
  opaque metadata<0..65535>;
  opaque something_else<0..255>;
} LogRootV2;
struct {
  Version version;
  select(version) {
    case v1: LogRootV1;
    case v2: LogRootV2;
  }
} LogRoot;
```

(might well be overkill though)

Reply:
Great idea :-)